dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPI.java

```diff
@@ -429,8 +429,38 @@ Folder createFolders(String path, Host host, User user, boolean respectFrontEndP
 	 * @throws DotDataException
 	 * @throws DotSecurityException
 	 */
-	boolean move(Folder folderToMove, Host newParentHost,User user,boolean respectFrontEndPermissions)throws DotDataException, DotSecurityException;
+	boolean move(Folder folderToMove, Host newParentHost,User user,boolean respectFrontEndPermissions) throws DotDataException, DotSecurityException;
 
+
+	/**
+	 * Move the currentFolder to non-existing folder.
+	 * @param newFolder     {@link String} destination
+	 * @param currentFolder {@link Folder} folder to move to the destination
+	 * @param user			{@link User}
+	 * @param respectFrontendRoles {@link Boolean}
+	 * @return boolean true if successfully removed
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
+	boolean moveWhenDestinationDoesNotExists(final String newFolder,
```

Comment:
Can we make these methods private in the folder API and just have the move method call them as needed?  I feel like we are breaking encapsulation forcing decisions on API users that they should not have to make.

Also, this private method should be called:
`moveWhenDestinationDoesNotExist` singular or just 'moveToNewDestination`

Reply:
ok